### ISSUE TITLE
Multiplication star is modified like a pointer star in for loops

### ISSUE THREAD

**Issue by Ambrevar** [uncrustify/uncrustify#399]
In 0.61 with this config:

```
sp_arith = force
sp_after_ptr_star = remove
```

the following should not change:

```
for(i = 0; i * i < n; i++)
```
but I get

```
for(i = 0; i *i < n; i++)
```
as if `i` were a pointer.

**Comment by kankaristo** [comment]
This happens even if one side is an integer:
```c++
for (int i = 0; i * 2 < 100; ++i) {}
```

Gets changed to:
```c++
for (int i = 0; i* 2 < 100; ++i) {}
```

But if you flip it the other way around, it stays the same:
```c++
for (int i = 0; 2 * i < 100; ++i) {}
```

Config for the above result:
```
sp_arith = add
sp_before_ptr_star = remove
```

**Comment by bengardner** [comment]
Fixed in 692d227.
